proposals/FileSystemSubscriptionManager.md

```diff
@@ -0,0 +1,125 @@
+# Proposal: FileSystemSubscriptionManager
+
+## Authors
+
+* [Ming-Ying Chung](https://github.com/mingyc) (Google)
+* [Austin Sullivan](https://github.com/a-sully) (Google)
+
+## Background
+
+The [FileSystemObserver API][fso-api] allows browsing context to receive records of changes for the observed files or directories. However, changes which occur while the website has no open tabs are not visible to the website.
+
+This document proposes integrating file system change observing with service worker registration, such that a service worker may be woken up on a change to the local file system.
+
+[fso-api]: https://chromestatus.com/feature/4622243656630272
+
+## Goal
+
+Allow web applications that do not have an open tab to quickly respond to changes to registered files and directories on the local file system.
+
+## Discussion
+
+### Option 1. Updating FileSystemObserver API to outlive Service Worker
+
+#### Enabling FileSystemObserver in Service Worker
+
+By [current design][fso], an instance of FileSystemObserver should only report changes which occur while the observer is connected and the website has an open & active tab.
+
+Enabling creations of FileSystemObserver in ServiceWorker will tie its lifetime with ServiceWorkerGlobalScope. Users might expect such observers to continue to watch files in the background. However, in browser implementation, service workers that haven’t received new events in a certain period of time, e.g. [30s in Chrome][chrome-sw], will likely be terminated.
+
+Hence, simply enabling it doesn't satisfy the goal. This approach might only work for websites that either can ensure a long-running service worker or are not really interested in using service worker registration for observing the file change events after the websites are closed.
+
+[fso]: https://github.com/whatwg/fs/blob/main/proposals/FileSystemObserver.md#handling-changes-made-outside-the-lifetime-of-a-filesystemobserver
+[chrome-sw]: https://developer.chrome.com/blog/longer-esw-lifetimes#background
+
+#### Outliving Service Worker
+
+What if updating FileSystemObserver to allow it to outlive without the limit of running in ServiceWorkerGlobalScope?
+
+If such an option is implemented, there needs to be mechanisms to
+Tell when the browser should stop watching file changes.
+Handle file changes happen after the service worker is already terminated but the website is still open. Possibly needs a way to wake up a new service worker.
+
+There are existing mechanisms to auto wake up new service workers. Hence the next option.
+
+### Option 2. Utilizing Service Worker Registration
```

Comment:
I think it would be good to mention how this pattern has precedents, e.g. push subscription manager and probably some others.

Reply:
Done.